### ISSUE TITLE
Why does the representation model affect topic reduction? 

### ISSUE THREAD

**Issue by zilch42** [MaartenGr/BERTopic#1104]
Hi Maarten,

I have a question about `reduce_topics`. It would seem that I get a different result from `reduce_topics(nr_topics='auto')` depending on which representation model I'm using (and I am setting a UMAP seed so the results are reproducible between runs). My understanding from reading the documentation was that `reduce_topics` works by clustering the ctfidf embeddings which are calculated before the topic representation, and indeed recalculated in `reduce_topics`, so why would the representation model have an influence? See example below:

```
from sklearn.datasets import fetch_20newsgroups
from bertopic import BERTopic
from bertopic.representation import MaximalMarginalRelevance, KeyBERTInspired
from bertopic.vectorizers import ClassTfidfTransformer
from sentence_transformers import SentenceTransformer
from sklearn.feature_extraction.text import CountVectorizer
import numpy as np
from umap import UMAP

data = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))
docs = data['data']

embedding_model = SentenceTransformer("all-MiniLM-L6-v2")
embeddings = embedding_model.encode(docs, show_progress_bar=True)

umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42)
ctfidf_model = ClassTfidfTransformer()
vectorizer_model = CountVectorizer(stop_words="english", ngram_range=(1, 2))
representation_model = MaximalMarginalRelevance(diversity=0.3)

topic_model = BERTopic(embedding_model=embedding_model, 
                       umap_model=umap_model, 
                       ctfidf_model=ctfidf_model, 
                       vectorizer_model=vectorizer_model, 
                       representation_model=representation_model)
topics, probs = topic_model.fit_transform(docs, embeddings)

topic_labels = topic_model.generate_topic_labels(separator=', ')
topic_model.set_topic_labels(topic_labels)

topic_model.get_topic_info()
```

Initial topics with MMR (219)
![image](https://user-images.githubusercontent.com/39004650/226223256-32ea9ec4-101e-4bec-aa3b-c5c2b834cff9.png)

```
topic_model.reduce_topics(docs, nr_topics="auto")
topic_model.get_topic_info()
```

Reduced topics with MMR (150)
![image](https://user-images.githubusercontent.com/39004650/226223269-e2e77f1b-7da8-4cb1-8de3-5393e4292ab2.png)

Retrain the model from scratch using KeyBERTInspired representation 
```
representation_model = KeyBERTInspired()

topic_model = BERTopic(embedding_model=embedding_model, 
                       umap_model=umap_model, 
                       ctfidf_model=ctfidf_model, 
                       vectorizer_model=vectorizer_model, 
                       representation_model=representation_model)
topics, probs = topic_model.fit_transform(docs, embeddings)

topic_labels = topic_model.generate_topic_labels(separator=', ')
topic_model.set_topic_labels(topic_labels)

topic_model.get_topic_info()
```

Initial topics with KeyBERTInspired (219)
![image](https://user-images.githubusercontent.com/39004650/226223285-287dc2b4-a384-48bf-b448-6978d345fd57.png)

```
topic_model.reduce_topics(docs, nr_topics="auto")
topic_model.get_topic_info()
```

Reduced topics with KeyBERTInspired (145)
![image](https://user-images.githubusercontent.com/39004650/226223299-b6fad32a-e42d-45fa-a7c1-a47465f275f7.png)

**Comment by MaartenGr** [comment]
When the topics are reduced, they are reduced based on the topic embeddings of each topic. These topic embeddings are created from the top n words per topic. You can find a bit more about the procedure here:

https://github.com/MaartenGr/BERTopic/blob/7d45cf710aefff178d54403989e90432e0c49910/bertopic/_bertopic.py#L3086-L3095

As a result, if you use a different representation model, then that will change those words which in turn will create different topic embeddings.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
